starter-core/src/main/java/io/micronaut/starter/feature/dekorate/DekorateFeature.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.dekorate;
+
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.OneOfFeature;
+
+/**
+ * @author Pavol Gressa
+ * @since 2.0
+ */
+public interface DekorateFeature  extends OneOfFeature {
+
+    @Override
+    default Class<?> getFeatureClass() {
+        return DekorateFeature.class;
+    }
+
+    @Override
+    default boolean supports(ApplicationType applicationType) {
+        return true;
+    }
```

Comment:
In all the classes that extend this you have:

```java
return applicationType == ApplicationType.DEFAULT || applicationType == ApplicationType.GRPC;
```

You could move that statement here and remove the method `supports` in all those classes.

Reply:
Please also add tests about this.

Reply:
Yeah, I didn't know that draft PRs are subject for review :/ .. it wasn't definitely ready for review. I would say now it is ready  (except the versions in bom)

Reply:
yeah, sorry about that. I didn't notice it was a draft PR